### PR TITLE
@SectionBuilder

### PR DESCRIPTION
**Pull request by RomanPodymov**
Hello.
Thank you for Eureka.
Added the first `@resultBuilder` to the project. If you like it, I will add more.
@mats-claassen Or should I add some documentation for `@SectionBuilder`?

### CODE REVIEW

File: Source/Core/ResultBuilders.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public static func buildEither(second components: [RowsProvider]) -> [BaseRow] {
+        components.flatMap { $0.rows }
+    }
+
+    public static func buildOptional(_ components: [RowsProvider]?) -> [BaseRow] {
+        components?.flatMap { $0.rows } ?? []
+    }
+    
+    public static func buildExpression(_ expression: String?) -> [BaseRow] {
+        [.init(tag: expression)]
```
Comment (mats-claassen) [inline Source/Core/ResultBuilders.swift:61]: This would be instantiating a BaseRow (not a concrete Row). Not sure about how this would be useful

**Review by mats-claassen — COMMENTED** [review]
Hi, thanks for the contribution. Looks great overall, just one comment

**Comment by mats-claassen** [comment]
Also, yes please add documentation to the Readme about this feature

File: Source/Core/ResultBuilders.swift (hunk 1 above)
Comment (RomanPodymov) [reply Source/Core/ResultBuilders.swift]: Hello @mats-claassen 
Exactly, deleted it.

**Comment by RomanPodymov** [comment]
> Also, yes please add documentation to the Readme about this feature

Done.

**Review by mats-claassen — APPROVED** [review]

**Pull request merged**